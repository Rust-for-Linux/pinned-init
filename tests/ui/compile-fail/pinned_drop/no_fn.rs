use pin_init::*;

#[pin_data(PinnedDrop)]
struct Foo {}

#[pinned_drop]
impl PinnedDrop for Foo {}

fn main() {}
