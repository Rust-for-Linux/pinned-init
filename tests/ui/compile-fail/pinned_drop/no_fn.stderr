error[E0046]: not all trait items implemented, missing: `drop`
 --> tests/ui/compile-fail/pinned_drop/no_fn.rs:7:1
  |
7 | impl PinnedDrop for Foo {}
  | ^^^^^^^^^^^^^^^^^^^^^^^ missing `drop` in implementation
  |
  = help: implement the missing item: `fn drop(self: Pin<&mut Self>, _: OnlyCallFromDrop) { todo!() }`
